google-cloud-storage/src/test/java/com/google/cloud/storage/ITOpenTelemetryTest.java

```diff
@@ -58,29 +48,80 @@ public void checkInstrumentation() {
     StorageOptions storageOptions =
         StorageOptions.http().setOpenTelemetrySdk(openTelemetrySdk).build();
     Storage storage = storageOptions.getService();
-    storage.create(BucketInfo.of(generator.randomBucketName()));
+    String bucket = randomBucketName();
+    storage.create(BucketInfo.of(bucket));
     TestExporter testExported = (TestExporter) exporter;
     SpanData spanData = testExported.getExportedSpans().get(0);
     Assert.assertEquals("Storage", getAttributeValue(spanData, "gcp.client.service"));
     Assert.assertEquals("googleapis/java-storage", getAttributeValue(spanData, "gcp.client.repo"));
     Assert.assertEquals(
         "com.google.cloud.google-cloud-storage",
         getAttributeValue(spanData, "gcp.client.artifact"));
+    Assert.assertEquals("http", getAttributeValue(spanData, "rpc.system"));
+
+    // Cleanup
+    RemoteStorageHelper.forceDelete(storage, bucket);
   }
 
   @Test
-  public void noOpDoesNothing() {
-    StorageOptions storageOptions = StorageOptions.http().build();
+  public void checkInstrumentationGrpc() {
+    SpanExporter exporter = new TestExporter();
+
+    OpenTelemetrySdk openTelemetrySdk =
+        OpenTelemetrySdk.builder()
+            .setTracerProvider(
+                SdkTracerProvider.builder()
+                    .addSpanProcessor(SimpleSpanProcessor.create(exporter))
+                    .build())
+            .build();
+    StorageOptions storageOptions =
+        StorageOptions.grpc().setOpenTelemetrySdk(openTelemetrySdk).build();
     Storage storage = storageOptions.getService();
-    storage.create(BucketInfo.of(generator.randomBucketName()));
+    String bucket = randomBucketName();
+    storage.create(BucketInfo.of(bucket));
+    TestExporter testExported = (TestExporter) exporter;
+    SpanData spanData = testExported.getExportedSpans().get(0);
```

Comment:
Do you plan on testing which spans are collected per operation in addition to Span attributes? I think this would be more useful for Upload and Download ops.

Reply:
I don't know if i understand this question. what do you mean by which spans are collected

Reply:
ah I see, I haven't expanded this for that case yet but yeah we would want to test each span is generated. Maybe we could check for the size to make sure if we expect 3 writeObjectRequests we would have 3 spans

Reply:
It's a good start to assert span count; and expand from there.